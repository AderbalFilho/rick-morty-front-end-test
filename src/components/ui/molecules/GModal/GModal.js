import React from 'react';
import classNames from 'classnames/bind';
import PropTypes from 'prop-types';

import './GModal.scss';

function GModal({ children, classes, ...rest }) {
  const className = classNames('g-modal', classes);

  return (
    <div className={className} {...rest}>
      {children}
    </div>
  );
}

GModal.propTypes = {
  children: PropTypes.node,
  classes: PropTypes.string,
};

export default GModal;
