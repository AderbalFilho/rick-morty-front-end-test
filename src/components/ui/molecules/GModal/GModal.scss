.g-modal {
  align-items: center;
  background-color: var(--modal-background-color);
  display: flex;
  flex-direction: column;
  height: 100vh;
  justify-content: center;
  min-width: 100%;
  overflow-y: hidden;
  position: fixed;
  z-index: 1;
}
