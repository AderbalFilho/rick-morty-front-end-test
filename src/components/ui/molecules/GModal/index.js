import GModal from './GModal';

export default GModal;
