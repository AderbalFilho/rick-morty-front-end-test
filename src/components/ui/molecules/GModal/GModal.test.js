import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import GModal from './GModal';

describe('GModal', () => {
  it('matches snapshot', () => {
    const testRenderer = TestRenderer.create(<GModal>inGaia</GModal>);
    const jsonComponent = testRenderer.toJSON();

    expect(jsonComponent).toMatchSnapshot();
  });

  it('renders children', () => {
    const { container } = render(<GModal>inGaia</GModal>);
    const modal = container.querySelector('.g-modal');

    expect(modal.textContent).toBe('inGaia');
  });

  it('renders classes', () => {
    const { container } = render(
      <GModal classes="modal-loading">inGaia</GModal>
    );
    const modal = container.querySelector('.g-modal');

    expect(modal.classList).toContain('modal-loading');
  });
});
