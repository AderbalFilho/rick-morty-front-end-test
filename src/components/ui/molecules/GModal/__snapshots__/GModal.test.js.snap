// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GModal matches snapshot 1`] = `
<div
  className="g-modal"
>
  inGaia
</div>
`;
